src/image/image_type.cpp

```diff
@@ -7,6 +7,19 @@
 
 namespace turi{
 
+image_type::image_type(boost::gil::rgb8_image_t gil_image) {
+  auto it = view(gil_image).begin();
+  auto data = (uint8_t*) &boost::gil::at_c<0>(*it);
+  m_height = gil_image.height();
+  m_width = gil_image.width();
+  m_channels = boost::gil::num_channels<boost::gil::rgb8_image_t>();
+  m_image_data_size = m_height * m_width * m_channels;
+  m_version = IMAGE_TYPE_CURRENT_VERSION;
+  m_format = Format::RAW_ARRAY;
+  m_image_data.reset(new char[m_image_data_size]);
+  memcpy(&m_image_data[0], data, m_image_data_size);
```

Comment:
When possible, use the C++ std::copy(...) function instead of memcpy -- it's safer to use and does the correct thing for all types (memcpy may not).

Reply:
Done, and changed the traditional flex_image constructor to use std::copy as well. Let me know if my usage is okay!